utils/avro-components-utils/src/main/scala/pl/touk/nussknacker/engine/avro/encode/AvroSchemaOutputValidator.scala

```diff
@@ -88,6 +90,15 @@ class AvroSchemaOutputValidator(validationMode: ValidationMode) extends LazyLogg
     }
   }
 
+  private def validateUnionInput(union: TypedUnion, schema: Schema, path: Option[String]) = {
+    if (validationMode == ValidationMode.strict && !union.possibleTypes.forall(validateTypingResult(_, schema, path).isValid))
```

Comment:
Maybe better change it to positive result? I mean: 
```
if (validationMode == ValidationMode.strict && union.possibleTypes.forall(validateTypingResult(_, schema, path).isValid)) => valid
```
below the same and at the end return invalid?